presto-main/src/main/java/com/facebook/presto/operator/aggregation/TruncatedHistogram.java

```diff
@@ -0,0 +1,157 @@
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.facebook.presto.type.MapType;
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Ints;
+
+import java.lang.invoke.MethodHandle;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.metadata.Signature.comparableTypeParameter;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.BLOCK_INDEX;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.BLOCK_INPUT_CHANNEL;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.INPUT_CHANNEL;
+import static com.facebook.presto.operator.aggregation.AggregationMetadata.ParameterMetadata.ParameterType.STATE;
+import static com.facebook.presto.operator.aggregation.AggregationUtils.generateAggregationName;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.util.Failures.checkCondition;
+import static com.facebook.presto.util.Reflection.methodHandle;
+import static java.lang.String.format;
+
+public class TruncatedHistogram
+        extends SqlAggregationFunction
+{
+    public static final TruncatedHistogram TRUNCATED_HISTOGRAM = new TruncatedHistogram();
+    public static final String NAME = "truncated_histogram";
+    private static final MethodHandle OUTPUT_FUNCTION = methodHandle(TruncatedHistogram.class, "output", TruncatedHistogramState.class, BlockBuilder.class);
+    private static final MethodHandle INPUT_FUNCTION = methodHandle(TruncatedHistogram.class, "input", Type.class, TruncatedHistogramState.class, long.class, Block.class, int.class);
+    private static final MethodHandle COMBINE_FUNCTION = methodHandle(TruncatedHistogram.class, "combine", TruncatedHistogramState.class, TruncatedHistogramState.class);
+
+    public static final int EXPECTED_SIZE_FOR_HASHING = 10;
+
+    public TruncatedHistogram()
+    {
+        super(NAME, ImmutableList.of(comparableTypeParameter("K")), "map<K,bigint>", ImmutableList.of(StandardTypes.BIGINT, "K"));
+    }
+
+    @Override
+    public String getDescription()
+    {
+        return "Count the number of times each value occurs, mixing up counts for infrequent values after max buckets";
+    }
+
+    @Override
+    public InternalAggregationFunction specialize(Map<String, Type> types, int arity, TypeManager typeManager, FunctionRegistry functionRegistry)
+    {
+        Type keyType = types.get("K");
+        Type valueType = BigintType.BIGINT;
+        return generateAggregation(keyType, valueType);
+    }
+
+    private static InternalAggregationFunction generateAggregation(Type keyType, Type valueType)
```

Comment:
Since `valueType` is always `BIGINT` you can remove the parameter and just use `BIGINT` in the code